testing/execution.ts

```diff
@@ -248,7 +250,7 @@ export async function executeFormulaOrSyncFromCLI({
 
     const {formulaSpecification, chainedCommand} = getFormulaSpecAndChainedCommand(manifest, formulaName);
 
-    if (formulaSpecification.type === FormulaType.Sync) {
+    if (formulaSpecification.type === FormulaType.Sync && chainedCommand?.type !== FormulaType.Sync) {
```

Comment:
Sounds like this is already on your cleanup list, but I'd try to have one `if` case here for `if (formulaSpecification.type === FormulaType.Sync)` and handle the sub-cases inside of it, ideally using an exhaustive `switch` for the chained command types.